src/frontend/src/heartbeat.rs

```diff
@@ -53,7 +54,7 @@ impl HeartbeatTask {
         resp_handler_executor: HeartbeatResponseHandlerExecutorRef,
     ) -> Self {
         HeartbeatTask {
-            server_addr: opts.grpc.addr.clone(),
+            peer_addr: addrs::resolve_addr(&opts.grpc.addr, Some(&opts.grpc.hostname)),
```

Comment:
In the example configuration file of datanode, there is a hostname https://github.com/GreptimeTeam/greptimedb/blob/22d12683b47e36aa349ca6d6ffe619444763f669/config/datanode.example.toml#L47
But it's not included in the frontend's configuration file. Can we add it?

Reply:
Fixed via https://github.com/GreptimeTeam/greptimedb/pull/4199/commits/d059e915dbc2ba6ec2f66592abf8e63b15c070a0